addons/crm_livechat/models/mail_channel.py

```diff
@@ -26,16 +26,32 @@ def _execute_command_lead(self, **kwargs):
         self._send_transient_message(partner, msg)
 
     def _convert_visitor_to_lead(self, partner, channel_partners, key):
+        """ Create a lead from channel /lead command
+        :param partner: internal user partner (operator) that created the lead;
+        :param channel_partners: channel members;
+        :param key: operator input in chat ('/lead Lead about Product')
+        """
         description = ''.join(
             '%s: %s\n' % (message.author_id.name or self.anonymous_name, message.body)
             for message in self.channel_message_ids.sorted('id')
         )
+        # if public user is part of the chat: consider lead to be linked to an
+        # anonymous user whatever the participants. Otherwise keep only share
+        # partners (no user or portal user) to link to the lead.
+        customers = self.env['res.partner']
+        for customer in channel_partners.partner_id.filtered('active'):
```

Comment:
Why the specific filter on "active" ?
When checking the code it doesn't seem to pass archived records.

Reply:
To add up to this question, isn't the public user especially archived? If so, this filter would lead to not detecting it at the next step.

Reply:
Yes, it was a first implementation: using is_public is better and active is actually not necessary. Could even be counterproductive is public partner is archived indeed.